### PR TITLE
Support of push register with Installation

### PR DESCRIPTION
**Pull request by dhei**
For #74 

Current push methods in Android SDK take in two forms: `register(pnsHandle)` and `register(pnsHandle, templates)`
We are adding an additional `register(Installation)` where Installation class reflects https://msdn.microsoft.com/en-us/library/azure/microsoft.azure.notificationhubs.notificationhubclient.createorupdateinstallation.aspx

### CODE REVIEW

File: e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/MainActivity.java [hunk 1]
```diff
@@ -183,7 +180,6 @@ public void run() {
                         adapter.add(new RoundTripTests());
                         adapter.add(new QueryTests());
                         adapter.add(new UpdateDeleteTests());
-                        //adapter.add(new ClientSDKLoginTests());
```
Comment (pvbchak) [inline e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/MainActivity.java]: > //adapter.add(new ClientSDKLoginTests()); [](start = 23, length = 42)

Is Removing ClientSDKLoginTests() intended?

File: sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/Installation.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    /**
+     * Registration Id, token or URI obtained from platform-specific notification service
+     */
+    public String pushChannel;
+
+    /**
+     * Installation expired or not
+     */
+    public boolean pushChannelExpired;
```
Comment (pvbchak) [inline sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/Installation.java]: >public boolean pushChannelExpired; [](start = 4, length = 34)

Is it required in registration scenario?

File: sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/Installation.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+     * @param platform platform
+     * @param pushChannel registration Id, token or URI obtained from platform-specific notification service
+     * @param pushChannelExpired installation expired or not
+     * @param pushVariables a collection of push variables
+     * @param tags a list of tags
+     * @param templates a collection of templates
+     */
+    public Installation(Date expirationTime, String installationId, String platform, String pushChannel,
+                        boolean pushChannelExpired, HashMap<String, String> pushVariables, List<String> tags, HashMap<String, InstallationTemplate> templates) {
+        this.expirationTime = expirationTime;
```
Comment (pvbchak) [inline sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/Installation.java]: Optional parameters (expirationTime & pushChannelExpired) should be last parameters.

File: e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/MainActivity.java (hunk 1 above)
Comment (dhei) [inline e2etest/src/main/java/com/microsoft/windowsazure/mobileservices/zumoe2etestapp/MainActivity.java]: Yes, ClientSDKLoginTests.java for client login flow scenario is completely broken. I don't have time to fix it now, opened an issue [here ](https://github.com/Azure/azure-mobile-apps-android-client/issues/92)to track it.

File: sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/Installation.java (hunk 3 above)
Comment (dhei) [inline sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/Installation.java]: Fixed

File: sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/Installation.java (hunk 2 above)
Comment (dhei) [inline sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/Installation.java]: expirationTime and pushChannelExpired are optional parameters, I've added documentation for it.

**Pull request merged**